### ISSUE TITLE
Stravalib 2.0 -> Introduce custom Pydantic types for field conversions

### ISSUE THREAD

**Issue by jsamoocha** [stravalib/stravalib#500]
- [x] Introduce custom types that inherit from the generated model's base types and add accessor logic to get the (Pint) `Quantity` representation of these base types.
- [x] Introduce custom type for time zones
- [x] Introduce custom type for time delta

Depends on #498

**Comment by lwasser** [comment]
this i think answers my question from the review about what the plan is.

**Comment by jsamoocha** [comment]
Yes, the plan is to access the "extended" attributes as follows:

```python
d = some_object.distance  # returns float
q = some_object.distance.quantity  # returns pint.Quantity (in meters)
z = unithelper.feet(some_object.distance)  # returns pint.Quantity (in feet)
```

I found a way to achieve this and maintain a correct type hierarchy.

**Comment by lwasser** [comment]
@jsamoocha this sounds great. 

i have a very basic question. The extender approach sounds like a smooth way to implement these conversions. I"m curious how this compares to returning the distance in quantities via a validator for each attr that forces the conversion when the model is populated. I have a few guesses as to why the extender is better but wanted to get your feedback! 

* perhaps involving typing and type overrides or 
* perhaps it would make the overall code base more complex having so many validators?
* Or finally perhaps it allows other users who may want to raw response values to convert as they please + the quantity conversions that are so nice to have in place.

**Comment by jsamoocha** [comment]
Closed by #534
